seismicpro/src/refractor_velocity/utils.py

```diff
@@ -86,3 +87,68 @@ def load_refractor_velocity_params(path, encoding="UTF-8"):
         params['coords'] = Coordinates(names=tuple(row[:2]), coords=tuple(row[2:4].astype(int)))
         params_list.append(params)
     return params_list
+
+# calculate number of refractors
+
+def binarization_offsets(offsets, times, step=20):
+    """Binarize offsets-times points."""
+    bins = np.arange(0, offsets.max() + step, step=step)
+    mean_offsets = np.arange(bins.shape[0]) * step + step / 2
+    mean_time = np.full(shape=bins.shape[0], fill_value=np.nan)
+    indices = np.digitize(offsets, bins, right=True)
+    for idx in np.unique(indices):
+        mean_time[idx] = times[idx == indices].mean()
+    nan_mask = np.isnan(mean_time)
+    return mean_offsets[~nan_mask], mean_time[~nan_mask]
+
+def calc_max_refractors_rv(offsets, times, min_offsets_diff, min_velocity_diff, start_refractor=1,
+                           max_refractors=10, init=None, bounds=None, weathering=False,
+                           plot_last=False):  # name and plot_last is debug features
+    """Calculate RefractorVelocity which have maximum number of refractor based on given constraints.
+    """
+    #pylint: disable-next=import-outside-toplevel
+    from .refractor_velocity import RefractorVelocity
+    name = ""   # debug feature
+    rv = None
+    for refractor in range(start_refractor, max_refractors + 1):
+        min_refractor_size = np.repeat(min_offsets_diff, refractor)
+        if weathering:
+            min_refractor_size[0] = 1
+        if offsets.max() > min_refractor_size[-1] * refractor:
+            rv_last = RefractorVelocity.from_first_breaks(offsets, times, n_refractors=refractor, init=init, tol=1e-6,
+                          bounds=bounds, min_velocity_step=min_velocity_diff, min_refractor_size=min_refractor_size)
+            # rv_last.plot(title=rv_last.fit_result.fun)
+        else:
+            break
+        n_points, _ = np.histogram(rv_last.offsets, bins=rv_last.piecewise_offsets)
+        if (n_points > 1).all() and (rv is None or rv_last.fit_result.fun < rv.fit_result.fun):
+            rv = rv_last
+        else:
+            break
+    if plot_last and rv is not None:  # debug feature
+        rv.plot(title=name)
+    return rv
+
+def calc_optimal_velocity(survey, min_offsets_diff=300, min_velocity_diff=300, fb_col=HDR_FIRST_BREAK, weathering=True,
+                     name=None, plot_last=False):
+    """Calculate one velocity model describe passed survey."""
+    if len(survey.indices) < 1:
```

Comment:
`survey` has `__len__`:
```suggestion
    if len(survey) < 1:
```

Reply:
`survey.n_gathers` prefer here.